scripts/rpc-maas-tool.py

```diff
@@ -13,199 +13,575 @@
-                    error = 1
-                if available is False:
-                    print(' - Check %s (%s) did not run correctly' %
-                          (check.id, check.label))
-                elif status not in ('okay', 'success'):
-                    print(" - Check %s (%s) ran correctly but returned a "
-                          "'%s' status" % (check.id, check.label, status))
-
-
-def delete(args, conn):
-    count = 0
-
-    if args.force is False:
-        print("*** Proceeding WILL delete ALL your checks (and data) ****")
-        if raw_input("Type 'from orbit' to continue: ") != 'from orbit':
-            return
-
-    for entity in _get_entities(args, conn):
-        for check in conn.list_checks(entity):
-            conn.delete_check(check)
-            count += 1
-
-    print("Number of checks deleted: %s" % count)
-
-
-def remove_defunct_checks(args, conn):
-    check_count = 0
-
-    for entity in _get_entities(args, conn):
-        for check in conn.list_checks(entity):
-            if re.match('filesystem--.*', check.label):
-                conn.delete_check(check)
-                check_count += 1
-
-    print("Number of checks deleted: %s" % check_count)
-
-
-def remove_defunct_alarms(args, conn):
-    alarm_count = 0
-    defunct_alarms = {'rabbit_mq_container': ['disk_free_alarm', 'mem_alarm'],
-                      'galera_container': ['WSREP_CLUSTER_SIZE',
-                                           'WSREP_LOCAL_STATE_COMMENT']}
-
-    for entity in _get_entities(args, conn):
-        for alarm in conn.list_alarms(entity):
-            for container in defunct_alarms:
-                for defunct_alarm in defunct_alarms[container]:
-                    if re.match('%s--.*%s' % (defunct_alarm, container),
-                                alarm.label):
-                        conn.delete_alarm(alarm)
-                        alarm_count += 1
-
-    print("Number of alarms deleted: %s" % alarm_count)
-
+from rackspace_monitoring.drivers import rackspace
+from rackspace_monitoring.providers import get_driver
+from rackspace_monitoring.types import Provider
 
-def _get_conn(config, driver):
-    conn = None
+DEFAULT_CONFIG_FILE = '/root/.raxrc'
+logging.basicConfig(level=logging.DEBUG,
+                    datefmt="",
+                    format="%(message)s",
+                    stream=sys.stdout)
+LOGGER = logging.getLogger(__name__)
+
+
+class RpcMaas(object):
+    """Class representing a connection to the MAAS Service"""
+
+    def __init__(self, entity_prefix='', config_file=DEFAULT_CONFIG_FILE):
+        self.entity_prefix = entity_prefix
+        self.config_file = config_file
+        self.driver = get_driver(Provider.RACKSPACE)
+        self._get_conn()
+        self._get_overview()
+        self._add_links()
+        self.entities = [e['entity'] for e in self.overview
+                         if entity_prefix in e['entity'].label]
```

Comment:
If you want to use a prefix then in should probably be replaced with the startswith string method. My preference would be to modify the tool to take a list of entities. Ansible already know the names and it is easy enough to supply them to the script so that the command is exact. The removes the potential for an erroneous match to the prefix.